presqt/targets/github/functions/fetch.py

```diff
@@ -28,23 +32,39 @@ def github_fetch_resources(token):
         }
     """
     try:
-        validation_check(token)
+        header, username = validation_check(token)
     except PresQTResponseException:
         raise PresQTResponseException('The response returned a 401 unauthorized status code.',
                                       status.HTTP_401_UNAUTHORIZED)
 
-    data = github_paginated_data(token)
-
-    resources = []
-    for repo in data:
-        resource = {
-            "kind": "container",
-            "kind_name": "repo",
-            "container": None,
-            "id": repo["id"],
-            "title": repo["name"]}
-        resources.append(resource)
-
+    if search_parameter:
+        search_parameters = search_parameter['title'].replace(' ', '+')
+        search_url = "https://api.github.com/search/repositories?q={}+in:name".format(
+            search_parameters)
+        data = requests.get(search_url, headers=header).json()
+
+        resources = []
+        for repo in data['items']:
+            resource = {
+                "kind": "container",
+                "kind_name": "repo",
+                "container": None,
+                "id": repo["id"],
+                "title": repo["name"]}
+            resources.append(resource)
+
+    else:
+        data = github_paginated_data(token)
+
+        resources = []
```

Comment:
It looks like you may be able to combine this for loop with the one on 47.

You can set data_items in both the if and else and then just loop through that afterwards.

Reply:
yee, good call